cfe_internal/enterprise/CFE_knowledge.cf

```diff
@@ -34,93 +34,114 @@ bundle agent cfe_internal_setup_knowledge
+      depth_search => recurse_basedir("inf"),
       handle => "cfe_internal_setup_knowledge_files_doc_root_api",
       file_select => cfe_internal_exclude_sh_pl_scripts,
-      depth_search => recurse_basedir_exclude("inf"),
-      perms => mog("0644","root","root");
+      perms => mog("0440", "root", $(def.cf_apache_group) ),
+      comment => "No one should be able to write to the application config code,
+                  and only the webserver needs access to read the config.";
+
+      "$(cfe_internal_hub_vars.docroot)/api/."
+      depth_search => recurse_basedir("inf"),
+      handle => "cfe_internal_setup_knowledge_files_doc_root_api_scripts",
+      file_select => cfe_internal_sh_pl_scripts,
+      perms => mog("0550", "root", $(def.cf_apache_group) ),
+      comment => "The scripts need to be executable, but only by the
+                  webserver and root users.";
 
 
       "$(cfe_internal_hub_vars.docroot)/api/static/."
-      comment => "Ensure permissions for $(cfe_internal_hub_vars.docroot)/api/static (writing exported reports PDF/CSV)",
-      handle => "cfe_internal_setup_knowledge_files_doc_root_api_static",
-      depth_search => recurse_basedir("inf"),
-      perms => mog("0644",$(def.cf_apache_user),$(def.cf_apache_group));
+      perms => mog("0660", "root", $(def.cf_apache_group) ),
+      comment => "This is where exported PDF and CSV reports from Mission
+                  Portal are written, it be writeable by the webserver";
 
-      "$(cfe_internal_hub_vars.docroot)/api/dc-scripts"
-      comment => "Ensure permissions for $(cfe_internal_hub_vars.docroot)/api/dc-scripts (execute git push/pull)",
-      handle => "cfe_internal_setup_knowledge_files_doc_root_api_dc_scripts",
-      file_select => cfe_internal_sh_pl_scripts,
-      depth_search => recurse_basedir("inf"),
-      perms => mog("0755","root","root");
+      "$(cfe_internal_hub_vars.docroot)/api/static/."
+      comment => "Exported reports only need to be readable by the webserver.",
+      handle => "cfe_internal_setup_knowledge_files_doc_root_api_static",
+      depth_search => recurse("inf"),
+      file_select => cfe_internal_docroot_api_static_perms,
+      perms => mog("0440", "root", $(def.cf_apache_group) );
 
       "$(cfe_internal_hub_vars.docroot)/sql_lite/."
       comment => "Create a directory sql_lite for analytic to be working",
       handle => "cfe_internal_setup_knowledge_files_doc_root_sql_lite",
       create => "true",
       depth_search => recurse_basedir("inf"),
-      perms => mog("0644",$(def.cf_apache_user),$(def.cf_apache_group));
+      perms => mog("0640", "root", $(def.cf_apache_group));
 
       "$(sys.workdir)/httpd/logs/."
       comment => "Ensure permissions for $(sys.workdir)/httpd/logs",
       handle => "cfe_internal_setup_knowledge_files_httpd_logs",
       create => "true",
-      perms => mog("0664","root","cfapache");
+      perms => mog("0660", "root", $(def.cf_apache_group));
```

Comment:
Need to sync with /application/logs/. above.
0660 should be good, though, restricting root to readonly doesn't really buy much security..